### DISCUSSION TITLE
Client cannot connect "Network Timeout occurred... Mesh Server Connection Error"

### DISCUSSION THREAD

**Discussion by oswaldo-be** [Ylianst/MeshCentral#6546] [Issues]
Hej there,

I set up a nginx reverse proxy to manage mesh central.
So I can manage my machines as well as the machines of friends or family.

I could send the agent to my fathers computer, which is abroad and his computer connected right away. (win11 latest updates, no other firewall or anti-virus, not connected via VPN to my network..)
Then i tried the computer of my sister, but this client does show this unfortunate error above.
I thought maybe her router is set to IPv6 only, but it aint. It is set as my router to prefer IPv4.

my json
```
{
  "$schema": "https://raw.githubusercontent.com/Ylianst/MeshCentral/master/meshcentral-config-schema.json",
  "__comment1__": "This is a simple configuration file, all values and sections that start with underscore (_) are ignored. Edit a section and remove the _ in front of the name. Refer to the user's guide for details.",
  "__comment2__": "See node_modules/meshcentral/sample-config-advanced.json for a more advanced example.",
  "settings": {
    "Cert": "thedomain.my",
    "_WANonly": true,
    "_LANonly": true,
    "_sessionKey": "MyReallySecretPassword1",
    "TLSOffload": "IP of reverse proxy",
    "port": 4430,
    "aliasPort": 443,
    "redirPort": 800,
    "BrowserPong": 120,
    "AgentPong": 120,
    "WebRTC": "false",
    "_redirAliasPort": 80
  },
  "domains": {
    "": {
      "_title": "MyServer",
      "_title2": "Servername",
      "_minify": true,
      "_newAccounts": true,
      "_userNameIsEmail": true,
      "certUrl": "IP of reverse proxy"
    }
  },
  "_letsencrypt": {
    "__comment__": "Requires NodeJS 8.x or better, Go to https://letsdebug.net/ first before trying Let's Encrypt.",
    "email": "myemail@mydomain.com",
    "names": "myserver.mydomain.com",
    "skipChallengeVerification": true,
    "production": false
  }
}
```

Maybe someone can give my a hint...
Thank you very much in advance

**Comment by si458** [comment]
it depends what OS is your sisters computers and also what version as you dont say
for example Windows 11 23h2 works, but windows 11 24h2 doesnt
to get it to work on windows 11 24h2 you need to install the WMIC add on first then run the agent and install it

**Reply by oswaldo-be** [reply]
oh damn, thank you very much. 
Yes it could be 24h2 very well.
Will try

**Reply by oswaldo-be** [reply]
damn
![grafik](https://github.com/user-attachments/assets/545417d7-0091-401c-8d7a-af0ef6de72da)
it seems to be installed already
but she is also still running 23H2

**Reply by si458** [reply]
ok so can u share a screenshot of the error? where are you seeing the error exactually?
as with windows you generally dont see any error messages

**Reply by oswaldo-be** [reply]
Well I stopped the agent, went in the windows command line and started meshagent connect.
There I could see for the first time why the agent does not appear in my meshcentral
![grafik](https://github.com/user-attachments/assets/ad1836d6-2748-49fa-89c6-ad3eeddbb80c)

**Reply by si458** [reply]
an right ok! so what does it show exactally? (hide your server url if you must)

**Reply by si458** [reply]
basically `Network Timeout occurred` explains it couldnt connect to your meshcentral,
can you visit your meshcentral inside her web browser and get the login page?

**Reply by oswaldo-be** [reply]
yes I can.
But when I try /agent.ashx from the machine I get 404.
![grafik](https://github.com/user-attachments/assets/789ff7e5-fd9c-4976-aa99-c1f32cc2f4da)

**Reply by si458** [reply]
thats ok, thats expected, so long as you can get the login page you should be ok
all looks ok with your config apart from `certUrl` which should be `https://mydomain.here.com`
then try restarting meshcentral,
but other than that, it would be a networking issue on your sisters comp, like maybe a web proxy or VPN ?

**Reply by oswaldo-be** [reply]
Yeah I thought so.
But all she has is a sophos vpn connection, but it isnt active right now.
Thank you very much for your help.

**Reply by si458** [reply]
so it might be a firewall then thats blocking the websocket traffic
one thing u could try is logging into the web ui from her laptop,
if u can login and see destops etc, then websockets is working and something is blocking access to your meshcentral

**Reply by oswaldo-be** [reply]
Since my father is with his Computer at home the client does not connect. But right now he is connected via another wifi and the client connects.
So I assume there must be a problem with the Router Fritz!Box from AVM.
I have to investigate in that direction I assume

**Reply by oswaldo-be** [reply]
I checked this, it did not work as well.
But when I start the assistant on the other side and activate it, this works.
Does this give a hint?
